### PR TITLE
For $ExpectType assertions, alphabetically sort unions and intersections

### PR DESCRIPTION
**Pull request by andy-ms**
Not a perfect solution to Microsoft/TypeScript#17944 as it operates on strings and thus can't delve into types easily, but should be enough to fix our test failures for now.
When this is merged I'll have to update all the tests on DT to align with this at the same time this is published.

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
There's going to be an immediate limitation with type literals containing unions (i.e. `{ a: number | string }`). Does that not come up in DefinitelyTyped?

**Comment by andy-ms** [comment]
I didn't see a lot, so added code to just anything with an object literal in it for now.

File: src/rules/expectRule.ts [hunk 1]
```diff
@@ -308,6 +309,27 @@ function getExpectTypeFailures(
 	}
 }
 
+function fixupType(s: string): string {
+	// Don't mess up `{ a: number | string }` into `string } | { a: number`
+	if (s.includes("{")) {
```
Comment (DanielRosenwasser) [inline src/rules/expectRule.ts]: or `<`, `(`, `[`

**Comment by DanielRosenwasser** [comment]
If we don't expect this to work perfectly, why not try to see if the current ones are identical, then try the adjustment as a fallback mechanism? Does that sound reasonable?

File: src/rules/expectRule.ts [hunk 2]
```diff
@@ -298,7 +298,12 @@ function getExpectTypeFailures(
 
 			const actual = checker.typeToString(type, /*enclosingDeclaration*/ undefined, ts.TypeFormatFlags.NoTruncation);
 			if (actual !== expected) {
-				unmetExpectations.push({ node, expected, actual });
+				console.log("!!!", actual);
```
Comment (fbartho) [inline src/rules/expectRule.ts]: Debug logs probably need to be cleaned up in this file?

**Pull request closed without merging**